src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile

```diff
@@ -26,7 +26,9 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 COPY scripts/sccache.sh /scripts/
 RUN sh /scripts/sccache.sh
 
-# using llvm-link-shared due to libffi issues -- see #34486
+# Using llvm-link-shared due to libffi issues -- see #34486
+# This requires disabling CI LLVM too.
+ENV NO_DOWNLOAD_CI_LLVM 1
```

Comment:
This confuses me - we already use dynamically linked llvm on Linux, right? Why would that require disabling downloads?

Reply:
Setting llvm-link-shared isn't currently allowed with downloaded LLVM, because on some platforms that's static. In this case on these two builders we don't want to set it anyway since we're choosing to build with LLVM 12 (distro provided) regardless. Updated comments.

Reply:
> Setting llvm-link-shared isn't currently allowed with downloaded LLVM, because on some platforms that's static.

This took me a minute to understand - we prevent this at runtime in bootstrap, regardless of platform, because it's not possible to configure.

Ok, the comment makes more sense to me now. I do think we should remove that hard error on platforms where the option you're setting is already true for the pre-compiled llvm, but doesn't need to block this change.

Reply:
Well, in any case, we don't want to download LLVM in these builders :) They're llvm-12 builders.

But yeah, I could see us fixing that to be OK with setting it if it *will* be the same, it's just annoying to maintain that knowledge in two places.